overlord/snapstate/snapmgr.go

```diff
@@ -270,15 +312,17 @@ func (m *SnapManager) doPrepareSnap(t *state.Task, _ *tomb.Tomb) error {
 		}
 		snapst.LocalRevision = revision
 		ss.Revision = revision
+
+	}
+
+	if ss.SideInfo != nil {
+		snapst.Candidate = ss.SideInfo
```

Comment:
I was expecting to see here logic similar to what you had in prepare-revert: if we have ss.Revision inside our Sequence, pick up its side info. That means we don't need Revert or SideInfo in SnapSetup.